labs/b5.md

```diff
@@ -0,0 +1,364 @@
+
+Tools also tend to overlap in functionality -- for example there are multiple tools that can display interface information or test connectivity.  When possible, it is a good idea to use multiple tools to cross-check one another.
+
+Note that when it comes to real world networks there are even more factors to consider that we haven't touched on like network security. For example, two machines can have a fully functioning connection but if one machine has been configured to drop all packets then it might seem as if they aren't connected. 
+
+So take the output of these tools with a grain of salt, they a means of narrowing down issues. It is important not to misinterpret outputs or jump to conclusions too quickly.
+
+1. `hostname`
+A simple and straightforward command that can display information about a host, IP addresses, FQDN, and etc.
+
+2. `ping`
+ Another simple command, most of the time you'll be using ping as a first step towards testing connectivity. If ping can't reach a host then there is likely an issue with connectivity.
+
+    Moreover ping also provides metrics for Round Trip Time (RTT) and packet loss.  Round trip time is defined as the time it takes for a response to arrive after sending the ping packet.  These can prove to be very useful statistics.
+
+3. `traceroute`
+Traceroute sends packets Time to Live (TTL) equal to the number of hops. Routers decreases the value of TTL of an incoming packet and if it sees an incoming packet with TTL = 0 then drops it, otherwise it decreases the value and sends it further. At the same time it sends diagnosing information to the source about router's identity.
+
+    Traceroute provides a detailed view of the routers that a packet traverses while on its way to a destination.
+
+    If router does not respond within a timeout then traceroute prints an asterisk.
+
+4. `arp`
+Provides info on and the ability to manipulate the ARP cache of the system.
+
+    With arp you can display the system arp table.  Add, remove, or modify arp entries and much more.
+    
+5. `dig`
+Utility for doing dns query and triaging DNS issues.
+
+    Dig by default performs queries to nameservers in `/etc/resolv.conf` but some options allow you to: specify name server, choose query type (iterative vs recursive), and much more -- making `dig` a very flexible DNS tool.
+    
+6. `ip`
+`ip` is a command with many subcommands offering a lot of functionality -- so much it can be overwhelming at first. You will most commonly be using ip to display/modify routing, IP addresses, or network interfaces.
```

Comment:
I'd give more details since `ip` is such a critical command. e.g. mention what `ip addr`, `ip link`, `ip route` tell you

Reply:
Added some info about these common use cases